### DISCUSSION TITLE
Welcome to townsquare Discussions!

### DISCUSSION THREAD

**Discussion by mmeldo** [townteki/townsquare#139] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mmeldo** [comment]
Welcome contributors 👋 This is a first comment to break ice 🧊

**Comment by maikiro** [comment]
Hey, Thanks for having me, Im Chris or Maikiro from UK. :)

**Comment by stopfstedt** [comment]
Hey! I'm Stefan, I live in California. Currently contributing with card implementations mainly.

**Comment by mmeldo** [comment]
Hello guys, thanks for helping with this project :) And I am Milan, living In Czech Republic, but originally from Slovakia.

**Comment by bugladen** [comment]
Hello,

I am the owner/developer of the now defunct CCG Workshop (generic client/server architecture to play CCGs)  Had old Doomtown adapted at one point.
I am very interested in contributing to this project, both UI and engine.

**Reply by mmeldo** [reply]
Hi there and welcome.
Help would be very appreciated of course. The project is in a good place now, but there is still a ton of work waiting, especially in UI.
If you want to start with something smaller, there are card scripts, and of course some issues to be solved.
Please take a look at the https://github.com/townteki/townsquare/blob/master/docs/contributing.md which I did not update for some time, but should give you an idea.
If you will have any questions or want to discuss further, you will find me on Discord under `trimm#8444`
There is also a Doomtown server with the `development` channel that is used to discuss dev things for this client: https://discord.gg/TsCp3J4Y

Looking forward to your PRs :)

**Comment by bugladen** [comment]
Hello, hoping to get some tech help on server install.  Am somewhat versed in Linux, but npm and node are new to me.

Crashed and burned on the npm install.
Here is the output:

```
eddie@eddie-VirtualBox:~$ cd townsquare
eddie@eddie-VirtualBox:~/townsquare$ git submodule init

Submodule 'townsquare-json-data' (https://github.com/mmeldo/townsquare-json-data) registered for path 'townsquare-json-data'

eddie@eddie-VirtualBox:~/townsquare$ git submodule update
Cloning into '/home/eddie/townsquare/townsquare-json-data'...
Submodule path 'townsquare-json-data': checked out 'b4df9eeb2a277353a4e27c08bf0aa74d88ffbd19'

eddie@eddie-VirtualBox:~/townsquare$ npm install
npm ERR! Linux 4.15.0-163-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
npm ERR! typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
npm ERR! typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
npm ERR! typeerror     at /usr/share/npm/lib/install/deps.js:457:7
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
npm ERR! typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
npm ERR! typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
npm ERR! typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>
WARN engine cross-env@7.0.3: wanted: {"node":">=10.14","npm":">=6","yarn":">=1"} (current: {"node":"8.10.0","npm":"3.5.2"})
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
WARN engine config@3.3.6: wanted: {"node":">= 10.0.0"} (current: {"node":"8.10.0","npm":"3.5.2"})
WARN engine bcrypt@5.0.1: wanted: {"node":">= 10.0.0"} (current: {"node":"8.10.0","npm":"3.5.2"})

npm ERR! Please include the following file with any support request:
npm ERR!     /home/eddie/townsquare/npm-debug.log
eddie@eddie-VirtualBox:~/townsquare$ 
```

The only divergence from the instructions is instead of:
```
git clone https://github.com/townteki/townsquare.git
```
I forked and cloned from my fork.

**Reply by mmeldo** [reply]
I see you have node v8 on your box, but you need at least v10 (I recommend v14). If you have discord, we can discuss there as it would be quicker.
